_search-plugins/search-pipelines/ml-inference-search-response.md

```diff
@@ -388,4 +388,374 @@ The response confirms that the processor has generated text embeddings in the `p
     ]
   }
 }
-```
\ No newline at end of file
+```
+
+### Example: GENAI Use Case
+
+The following example shows you how to configure an `ml_inference` search response processor with a genai model and mapping the model response to the search extension.
+
+Step 0: Host a model
+The pre-requisite is a registered genai model in opensearch. For more information about externally hosted models, see [Connecting to externally hosted models]({{site.url}}{{site.baseurl}}/ml-commons-plugin/remote-models/index/). Here is a sample predict response using a registered model, which requires a prompt and a context field.
+
+```json
+POST /_plugins/_ml/models/EOF6wJIBtDGAJRTD4kNg/_predict  
+{
+  "parameters": {
+    "prompt":"\n\nHuman: You are a professional data analysist. You will always answer question based on the given context first. If the answer is not directly shown in the context, you will analyze the data and find the answer. If you don't know the answer, just say I don't know. Context: ${parameters.context.toString()}. \n\n Human: please summarize the documents \n\n Assistant:",
+    "context":"Dr. Eric Goldberg is a fantastic doctor who has correctly diagnosed every issue that my wife and I have had. Unlike many of my past doctors, Dr. Goldberg is very accessible and we have been able to schedule appointments with him and his staff very quickly. We are happy to have him in the neighborhood and look forward to being his patients for many years to come."
```

Comment:
minor: Can we tune the example to make it align with the prompt ? The prompt is for data analysis , this example is not.

Reply:
ok, I will use this cost calculation example instead

```
  "parameters": {
     "prompt":"\n\nHuman: You are a professional data analysist. You will always answer question: Which month had the lowest customer acquisition cost per new customer? based on the given context first. If the answer is not directly shown in the context, you will analyze the data and find the answer. If you don't know the answer, just say I don't know. Context: ${parameters.context.toString()}. \n\n Assistant:",
    "context":"Customer acquisition cost: January: $50, February: $45, March: $40. New customers: January: 500, February: 600, March: 750"
  }
```
the response will be
```
Based on the context provided:

- Customer acquisition cost in January was $50 and new customers were 500. So the cost per new customer was $50/500 = $0.10

- Customer acquisition cost in February was $45 and new customers were 600. So the cost per new customer was $45/600 = $0.075 

- Customer acquisition cost in March was $40 and new customers were 750. So the cost per new customer was $40/750 = $0.053

Therefore, the month with the lowest customer acquisition cost per new customer was March, as it had the lowest cost per customer of $0.053."""
 
```